docs/pages/machine-workload-identity/machine-id/deployment/kubernetes-oidc.mdx

```diff
@@ -0,0 +1,442 @@
+
+The examples in this guide will install a `tbot` deployment in the `default`
+Namespace of the Kubernetes cluster. Adjust references to `default` to the
+Namespace you wish to use.
+
+## Step 1/6. Verify support for Service Account Issuer Discovery
+
+Before continuing, verify that your Kubernetes cluster can issue tokens that
+Teleport will be able to verify. To do so, we'll want to perform the following
+steps:
+1. Fetch the cluster's OpenID configuration endpoint and extract the `issuer`
+   field
+2. Attempt to fetch the same OpenID configuration endpoint from its public
+   address, derived from the `issuer` field value, and extract the public
+   `jwks_uri`
+3. Attempt to fetch the `jwks_uri` to ensure Teleport will be able to do so when
+   your agents attempt to join
+
+To do so, ensure you have `kubectl`, `curl`, and `jq` available. Start by
+running this command to determine the public OIDC configuration URL:
+```code
+$ kubectl get --raw /.well-known/openid-configuration | jq -r '.issuer + "/.well-known/openid-configuration"'
+https://oidc.eks.us-west-2.amazonaws.com/id/$myClusterID/.well-known/openid-configuration
+```
+
+If this command fails, the Service Account Issuer Discovery feature is not
+enabled on your Kubernetes cluster.
+
+Next, we'll attempt to fetch the configuration document returned by the previous
+command. Note that this should be run over the public internet, such as your
+home internet connection, to help ensure the endpoint will be accessible to
+Teleport:
+```code
+$ curl https://oidc.eks.us-west-2.amazonaws.com/id/$myClusterID/.well-known/openid-configuration | jq
+{
+  "issuer": "https://oidc.eks.us-west-2.amazonaws.com/id/$myClusterID",
+  "jwks_uri": "https://oidc.eks.us-west-2.amazonaws.com/id/$myClusterID/keys",
+  "authorization_endpoint": "urn:kubernetes:programmatic_authorization",
+  "response_types_supported": [
+    "id_token"
+  ],
+  "subject_types_supported": [
+    "public"
+  ],
+  "claims_supported": [
+    "sub",
+    "iss"
+  ],
+  "id_token_signing_alg_values_supported": [
+    "RS256"
+  ]
+}
+```
+
+If this command succeeds, make a note of the `issuer` value for future steps.
+
+As one final check, you can also attempt to fetch the `jwks_uri`:
+```code
+$ curl https://oidc.eks.us-west-2.amazonaws.com/id/$myClusterID/keys
+{"keys":[{...snip...}]}
+```
+
+There's no need to record anything from this response - we just need to be
+certain that Teleport will be able to access this URL.
+
+<Admonition type="warning" title="OIDC Support">
+If any of these commands fail, additional steps may be needed to enable public
+[Service Account Issuer Discovery][said] is enabled for your Kubernetes
+provider.
+
+Refer to your cloud provider documentation to enable this feature, including:
+- [Amazon EKS](https://docs.aws.amazon.com/eks/latest/userguide/enable-iam-roles-for-service-accounts.html)
+- [Azure AKS](https://learn.microsoft.com/en-us/azure/aks/use-oidc-issuer)
```

Comment:
@hugoShaka did you notice if GKE required additional setup? I wasn't able to find any documentation about it, so maybe it's just always enabled out of the box?

Reply:
(Checked via DM, looks like it works out of the box on GKE)